tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -78,17 +80,78 @@ public void KillCursor_should_actually_work()
                 cursor = collection.FindSync("{}");
                 cursor.MoveNext();
 
-                var cursorId = ((AsyncCursor<BsonDocument>) cursor)._cursorId();
+                var cursorId = ((AsyncCursor<BsonDocument>)cursor)._cursorId();
                 cursorId.Should().NotBe(0);
                 cursor.Dispose();
 
                 var desiredResult = BsonDocument.Parse($"{{ \"cursorsKilled\" : [{cursorId}], \"cursorsNotFound\" : [], " +
                     $"\"cursorsAlive\" : [], \"cursorsUnknown\" : [], \"ok\" : 1.0 }}");
-                var result = ((CommandSucceededEvent) eventCapturer.Events[0]).Reply;
+                var result = ((CommandSucceededEvent)eventCapturer.Events[0]).Reply;
                 result.IsSameOrEqualTo(desiredResult);
             }
         }
 
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Tailable_cursor_should_be_able_to_be_cancelled_from_a_different_thread_with_expected_result(
+            [Values(false, true)] bool async)
+        {
+            string testCollectionName = "test";
+            string testDatabaseName = "test";
+            var client = CreateClient();
+            DropCollection(client, testDatabaseName, testCollectionName);
+            var database = client.GetDatabase(testDatabaseName);
+
+            var createCollectionOptions = new CreateCollectionOptions()
+            {
+                Capped = true,
+                MaxSize = 1000
+            };
+
+            database.CreateCollection(testCollectionName, createCollectionOptions);
+            var collection = database.GetCollection<BsonDocument>(testCollectionName);
+            collection.InsertOne(new BsonDocument());
+
+            var findOptions = new FindOptions<BsonDocument>()
+            {
+                BatchSize = 1,
+                CursorType = CursorType.Tailable
+            };
+
+            IAsyncCursor<BsonDocument> cursor;
+            if (async)
+            {
+                cursor = collection
+                    .FindAsync(
+                        FilterDefinition<BsonDocument>.Empty,
+                        findOptions)
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                cursor = collection
+                    .FindSync(
+                        FilterDefinition<BsonDocument>.Empty,
+                        findOptions);
+            }
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            var enumerator = cursor.ToEnumerable(cancellationTokenSource.Token).GetEnumerator();
+            enumerator.MoveNext();
+
+            var thread = new Thread(() =>
+            {
+                Thread.Sleep(200);
```

Comment:
Waiting for evergreen results to be sure that `200` is enough delay here

Reply:
Evergreen results show that 200 is not enough. Checking 500

Reply:
The problem wasn't related to the sleap dealy. However, I've rewrote this test in the way which is not depended on `Thread.Sleep`